flink-runtime/src/main/java/org/apache/flink/runtime/leaderelection/AbstractLeaderElectionService.java

```diff
@@ -27,29 +27,37 @@
 public abstract class AbstractLeaderElectionService implements LeaderElectionService {
 
     @Override
-    public LeaderElection createLeaderElection() {
-        return new DefaultLeaderElection(this);
+    public LeaderElection createLeaderElection(String contenderID) {
+        return new DefaultLeaderElection(this, contenderID);
     }
 
     /**
-     * Registers the given {@link LeaderContender} with the underlying {@code
-     * LeaderElectionService}. Leadership changes are starting to be reported to the {@code
-     * LeaderContender}.
+     * Registers the given {@link LeaderContender} under the passed {@code contenderID} with the
+     * underlying {@code LeaderElectionService}. Leadership changes are starting to be reported to
+     * the {@code LeaderContender}.
      */
-    protected abstract void register(LeaderContender contender) throws Exception;
+    protected abstract void register(String contenderID, LeaderContender contender)
+            throws Exception;
 
-    /** Removes the passed {@code LeaderContender} from the {@code LeaderElectionService}. */
-    protected abstract void remove(LeaderContender contender);
+    /**
+     * Removes the {@code LeaderContender} from the {@code LeaderElectionService} that is associated
+     * with the passed {@code contenderID}.
```

Comment:
inconsistent uses of "given" vs "passed".

Reply:
Maybe, it's my English here but I thought that I could use both terms interchangably. Anyway, I revisited the JavaDoc and removed both forms and rather referred to the parameter by their names explicitly.